### PR TITLE
contributing: add new section about commit log

### PR DESCRIPTION
**Pull request by emersion**
This is shamelessly stolen from Weston [1].

It's been a while we've transitioned away from merge commits and
work-style commit history, so it'd be nice to mention this in the
docs.

[1]: https://gitlab.freedesktop.org/wayland/weston/-/blob/main/CONTRIBUTING.md#formatting-and-separating-commits

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,6 +36,37 @@ your new features work correctly). Document all of the edge cases you're aware
...
+fix tests when disabled, redo broken doc formatting
+better transformed-view iteration (thanks Hannah!)
+try to catch more cases in tests
+tests: add new spline test
+fix compilation on splines
+doc: notes on reticulating splines
+compositor: add spline reticulation for view transforms
+```
+
+we aim to have a clean history which only reflects the final state, broken up
```
Comment (bl4ckb0ne) [inline CONTRIBUTING.md:57]: `we` -> `We`

**Comment by bl4ckb0ne** [comment]
Might be worth linking https://git-rebase.io/ for those who don't know about git history rewriting.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,6 +36,37 @@ your new features work correctly). Document all of the edge cases you're aware
 of so we can adequately test them - then verify the test plan yourself before
 submitting.
 
+## Commit Log
+
+Unlike many projects using GitHub and GitLab, Weston has a [linear, "recipe"
```
Comment (agx) [inline CONTRIBUTING.md:41]: s/Weston/wlroots/ ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,6 +36,37 @@ your new features work correctly). Document all of the edge cases you're aware
...
+
+```
+compositor: add spline reticulation for view transforms
+compositor: new iterator for view transforms
+tests: add view-transform correctness tests
+doc: fix formatting for view transforms
+```
+
+This ensures that the final patch series only contains the final state,
+without the changes and missteps taken along the development process.
```
Comment (agx) [inline CONTRIBUTING.md:68]: Maybe add some motivation like "to ease review, cherry-picks and reverts" ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -36,6 +36,37 @@ your new features work correctly). Document all of the edge cases you're aware
 of so we can adequately test them - then verify the test plan yourself before
 submitting.
 
+## Commit Log
+
+Unlike many projects using GitHub and GitLab, Weston has a [linear, "recipe"
+style](https://www.bitsnbites.eu/git-history-work-log-vs-recipe/) history. This
+means that every commit should be small, digestible, stand-alone, and
```
Comment (bl4ckb0ne) [inline CONTRIBUTING.md:43]: I found missleading to have the first example of what we should not do first.

**Review by agx — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (bl4ckb0ne) [reply CONTRIBUTING.md:43]: Fixed by @agx's comment about weston/wlroots

**Review by bl4ckb0ne — APPROVED** [review]
